src/Cake.Core/CakeEnvironment.cs

```diff
@@ -22,6 +22,9 @@ public DirectoryPath WorkingDirectory
             set { SetWorkingDirectory(value); }
         }
 
+        /// <inheritdoc/>
+        public ILaunchInfo LaunchInfo { get; }
```

Comment:
```suggestion
        /// <inheritdoc/>
        public DirectoryPath LaunchDirectory
        {
            get { return launchInfo.LaunchDirectory; }
        }
```

The `ILaunchInfo` is just an implementation detail to give you a way for you to store the `LaunchDirectory` early in the process, and inject into the places that need it. From the point of view of the API, for build scripts to access it, let's expose it in the same way that we do for `WorkingDirectory` above.